paasta_tools/api/views/instance.py

```diff
@@ -475,13 +500,92 @@ def marathon_smartstack_status(
     return smartstack_status
 
 
+def kubernetes_smartstack_status(
+    service: str,
+    instance: str,
+    job_config: kubernetes_tools.KubernetesDeploymentConfig,
+    service_namespace_config: ServiceNamespaceConfig,
+    pods: Sequence[V1Pod],
+    should_return_individual_backends: bool = False,
+) -> Mapping[str, Any]:
+
+    registration = job_config.get_registrations()[0]
+    instance_pool = job_config.get_pool()
+
+    smartstack_replication_checker = KubeSmartstackReplicationChecker(
+        nodes=kubernetes_tools.get_all_nodes(settings.kubernetes_client),
+        system_paasta_config=settings.system_paasta_config,
+    )
```

Comment:
So this seems like it'll only find kubernetes backends, yeah?  I think that is probably fine, but it seems like neither this nor the equivalent marathon service code is dealing with the case where we have instances in both mesos and kubernetes that are advertising using the same smartstack namespace.  Is that a problem?  Might be worth at least filing a ticket about.

Reply:
I think we won't have this case for a long time, such as more than a couple of hours, when we really start migration. I need to do some investigation about the migration plan.